src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs

```diff
@@ -1367,6 +1367,45 @@ public void TestNewArray()
             Assert.Null(ac.Initializer);
         }
 
+        [Fact]
+        public void TestNewPartialArray()
+        {
+            UsingTree("new partial[1]");
```

Comment:
This used to parse as a bad member declaration, as if it were `new partial` followed by an attribute list, which doesn't really make sense.

Reply:
Might want to check the behavior of `new file[1]` and `new required[1]` at the top level as well.